cmd_clean/main.go

```diff
@@ -53,10 +55,57 @@ func (t *Transport) base() http.RoundTripper {
 	return http.DefaultTransport
 }
 
+// Convert certificate in bytes array into stripped header for envoy sidecar
+func certPemBytesToHeader(caFile string) string {
+	certPem, err := os.ReadFile(caFile)
+	if err != nil {
+		log.Error(err, "failed to read ca file")
+		return ""
+	}
+	cert := string(certPem)
+	cert = strings.ReplaceAll(cert, "-----BEGIN CERTIFICATE-----", "")
+	cert = strings.ReplaceAll(cert, "-----END CERTIFICATE-----", "")
+	cert = strings.ReplaceAll(cert, "\n", "")
+	return strings.TrimSpace(cert)
+}
+
+// just a demo to show how to use customer http client
+// customer http client should handle verify and authentication
+// here using the basic user/password mode for authentication
+// not handling verify for none envoy client
+func createHttpClient(cf *config.NSXOperatorConfig, useExternalHttp bool) *http.Client {
+	var httpClient *http.Client = nil
+	if useExternalHttp {
+		tr := &http.Transport{}
+		transport := &Transport{Base: tr}
+		httpClient = &http.Client{
+			Transport: transport,
+		}
+
+		if cf.EnvoyPort != 0 {
```

Comment:
nit: let's make it consistent that we switch to Envoy mode as long as envoyHost is set, and default to port 1080 if unset.

Reply:
done